libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -457,13 +462,27 @@ export class TurnContext {
         });
 
         return this.emit(this._onSendActivities, output, () => {
-            return this.adapter.sendActivities(this, output)
-                .then((responses: ResourceResponse[]) => {
-                    // Set responded flag
-                    if (sentNonTraceActivity) { this.responded = true; }
-
-                    return responses;
+            if (this.activity.deliveryMode === 'buffered') {
+                // Append activities to buffer
+                const responses: ResourceResponse[] = [];
+                output.forEach((a) => {
+                    this.bufferedActivities.push(a);
+                    responses.push({ id: undefined });
                 });
+
+                // Set responded flag
+                if (sentNonTraceActivity) { this.responded = true; }
```

Comment:
Do we still send trace activities for this mode? The recipient of the bufferedActivities probably shouldn’t receive any traces.

Reply:
> Do we still send trace activities for this mode? The recipient of the bufferedActivities probably shouldn’t receive any traces.

Would we not want to send all activities back in bufferedActivites and then leave it up to the parent to decide what to? In the case that we are calling a skill, the skill could conceivably send trace activities back which we, in turn, want to be sent by the parent back to the emulator, for example.

Reply:
That’s true, on second thought I’d continue with the pattern that the parent and skill figure out what to do, rather than forcing behavior since the semantics are open-ended